### ISSUE TITLE
scatterjs-core@2.7.7 "There is no identity with an account set on your Scatter instance."

### ISSUE THREAD

**Issue by edwardfhsiao** [GetScatter/scatter-js#91]
Hi, @nsjames 

Initially getting an identity and forget it works well with ```scatterjs-core@2.7.7```, but getting an identity second time then forget it occurs ```Uncaught o {type: "There is no identity with an account set on your Scatter instance.", message: undefined, code: 423, isError: true}```

```js
import ScatterJS from 'scatterjs-core';

signOut(){
  console.log(ScatterJS.scatter.identity) // returns null
  ScatterJS.scatter.forgetIdentity().then(() => {
})
```
}

problem does not occur with ```scatterjs-core@2.3.8```

**Comment by nsjames** [comment]
Is this happening only with desktop, or only with extension, or both?

**Comment by edwardfhsiao** [comment]
on extension. I have not tested it with desktop yet. I am creating a repo to find out.

**Comment by nsjames** [comment]
I can't seem to reproduce this on either using the vanilla-eos mocksite actually.

**Comment by nsjames** [comment]
![image](https://user-images.githubusercontent.com/10360675/51965467-56a47180-2472-11e9-8134-5ba3c517eb55.png)

**Comment by nsjames** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/10360675/51965487-6623ba80-2472-11e9-9ac2-d20ebc461201.gif)

**Comment by nsjames** [comment]
In general though, you shouldn't be developing on the extension. The extension has been deprecated for almost 6 months, is no longer maintained, and has warnings inside of it urging users to update to desktop. It's very likely this works as intended on desktop but perhaps not the extension.

**Comment by edwardfhsiao** [comment]
confirmed not happening with desktop version

**Comment by edwardfhsiao** [comment]
https://github.com/edwardfhsiao/scatter-test-repo

```sh
npm i
```
then 
```sh
npm run server
```

0.0.0.0:3000

**Comment by nsjames** [comment]
So this might be a different issue actually.
Generally you only want to call `ScatterJS.connect` once, and then use it's API many times regardless of login status. 

What might be happening is that it's trying to reuse the same socket but opening another one along-side which is causing it to fail. 

Try moving the connect outside the scope and only running `getIdentity` in the login method

**Comment by edwardfhsiao** [comment]
sorry, that's my bad. By calling it only once does not make the difference. I updated my code

**Comment by nsjames** [comment]
Alright, so I noticed it only happens when you don't inject `{network}` into the connect. 
I pushed a fix to core (`scatterjs-core@2.7.9`) which fixed the issue. Thanks for the report! 👍
